scheduler/core/gocron_scheduler_test.go

```diff
@@ -276,3 +284,74 @@ func TestCatchUpAllSchedule(t *testing.T) {
 	catchupSuccess := g.CatchupAll(ctx, toTime)
 	assert.True(t, catchupSuccess)
 }
+
+func TestScheduleJob(t *testing.T) {
+	ctx := context.Background()
+	True := true
+	lastTime := time.Now()
+	scheduleFixed = models.SchedulableEntity{
+		BaseModel: adminModels.BaseModel{
+			UpdatedAt: time.Now(),
+		},
+		SchedulableEntityKey: models.SchedulableEntityKey{
+			Project: "project",
+			Domain:  "domain",
+			Name:    "fixed1",
+			Version: "version1",
+		},
+		FixedRateValue: 1,
+		Unit:           admin.FixedRateUnit_MINUTE,
+		Active:         &True,
+	}
+	t.Run("using schedule time", func(t *testing.T) {
+		wg := &sync.WaitGroup{}
+		wg.Add(1)
+		timedFuncWithSchedule := func(jobCtx context.Context, schedule models.SchedulableEntity, scheduleTime time.Time) error {
+			assert.Equal(t, lastTime, scheduleTime)
+			wg.Done()
+			return nil
+		}
+		c := cron.New()
+		g := GoCronScheduler{cron: c, jobStore: sync.Map{}}
+		err := g.ScheduleJob(ctx, scheduleFixed, timedFuncWithSchedule, &lastTime)
+		c.Start()
+		assert.NoError(t, err)
+		select {
+		case <-time.After(time.Minute * 2):
```

Comment:
does this have to be such a long interval for unit tests?

Reply:
This will 2 min only in failure case.This is more of an integration test since there is no easy way to unit test this particular behaviour

Reply:
Move this to integration test package and wont impact the unit tests